react/context/useShippingOption.ts

```diff
@@ -0,0 +1,275 @@
+        })
+
+        await updateSession(
+          country,
+          selectedZipcode,
+          coordinates,
+          pickup,
+          shippingMethod
+        )
+      }
+
+      dispatch({ type: 'SET_IS_LOADING', args: { isLoading: false } })
+    },
+    [account, dispatch]
+  )
+
+  useEffect(() => {
+    if (isSSR) {
+      return
+    }
+
+    const segmentZipCode = getFacetsData('zip-code')
+    const segmentCountryCode = getCountryCode()
+    const segmentShippingOption = getFacetsData('shipping') as ShippingMethod
+
+    dispatch({
+      type: 'SET_ZIPCODE',
+      args: { zipcode: segmentZipCode },
+    })
+
+    dispatch({
+      type: 'SET_SHIPPING_OPTION',
+      args: { shippingOption: segmentShippingOption },
+    })
+
+    dispatch({
+      type: 'SET_COUNTRY_CODE',
+      args: { countryCode: segmentCountryCode },
+    })
+
+    if (segmentZipCode) {
+      getAddress(segmentCountryCode, segmentZipCode, account).then((res) => {
+        dispatch({
+          type: 'SET_CITY',
+          args: { city: res.city },
+        })
+        dispatch({
+          type: 'SET_GEO_COORDINATES',
+          args: { geoCoordinates: res.geoCoordinates },
+        })
+        fetchPickups(
+          segmentCountryCode,
+          segmentZipCode,
+          res.geoCoordinates,
+          segmentShippingOption
+        )
+      })
+    } else {
+      dispatch({ type: 'SET_IS_LOADING', args: { isLoading: false } })
+    }
+  }, [account, isSSR, fetchPickups, dispatch])
+
+  const onError = (message: string) => {
+    dispatch({
+      type: 'SET_SUBMIT_ERROR_MESSAGE',
+      args: { submitErrorMessage: message },
+    })
+    dispatch({ type: 'SET_IS_LOADING', args: { isLoading: false } })
+
+    setTimeout(() => {
+      dispatch({
+        type: 'SET_SUBMIT_ERROR_MESSAGE',
+        args: { submitErrorMessage: undefined },
+      })
+    }, 3000)
+  }
+
+  const submitZipcode = async (selectedZipcode: string, reload = true) => {
+    if (!selectedZipcode) {
+      onError(intl.formatMessage(messages.postalCodeInputPlaceHolder))
+
+      return
+    }
+
+    if (!countryCode) {
+      return
+    }
+
+    dispatch({ type: 'SET_IS_LOADING', args: { isLoading: true } })
+
+    const { geoCoordinates: coordinates, city: cityName } = await getAddress(
+      countryCode,
+      selectedZipcode,
+      account
+    )
+
+    if (coordinates.length === 0) {
+      onError(intl.formatMessage(messages.postalCodeInputError))
+
+      return
+    }
+
+    const orderFormId = getOrderFormId()
+
+    if (orderFormId) {
+      await updateOrderForm(countryCode, selectedZipcode, orderFormId)
+    }
+
+    dispatch({ type: 'SET_CITY', args: { city: cityName } })
+
+    dispatch({
+      type: 'SET_GEO_COORDINATES',
+      args: { geoCoordinates: coordinates },
+    })
+
+    dispatch({
+      type: 'SET_ZIPCODE',
+      args: { zipcode: selectedZipcode },
+    })
+
+    await updateSession(countryCode, selectedZipcode, coordinates)
```

Comment:
Food for tought, isn't better to encapsulate all this fetch/update logic into an endpoint? Otherwise we will have to duplicate this when implementing it on FastStore.